giles-eco/src/main/java/edu/asu/diging/gilesecosystem/web/rest/FilesController.java

```diff
@@ -160,6 +164,88 @@
         return new ResponseEntity<String>(sw.toString(), HttpStatus.OK);
     }
 
+    @DocumentOwnerCheck
+    @RequestMapping(value = GET_DOCUMENT_PATH
+            + "/access/change", produces = "application/json;charset=UTF-8", method = RequestMethod.POST)
+    public ResponseEntity<String> changeDocumentAccess(@RequestParam(defaultValue = "") String accessToken,
+            HttpServletRequest request, @PathVariable("docId") String docId, @RequestParam("access") String access,
+            User user) {
+
+        if (docId == null || docId.isEmpty() || access == null || access.isEmpty()) {
```

Comment:
I'm fairly sure that if access isn't provided in the request, then Spring will throw an exception. Have you tested that?